### DISCUSSION TITLE
Issue to build an IPK

### DISCUSSION THREAD

**Discussion by pimdkrisf3** [bluenviron/mediamtx#1852] [Questions]
Hello

I want to create the ipk for an OpenWRT router (D-Link) according to the instructions. I always get the following error at point 4 ("make package/mediamtx/compile -j$(nproc)"):
```
......go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/pmezard/go-difflib v1.0.0
go: finding github.com/pion/udp/v2 v2.0.1
go: finding github.com/pion/srtp/v2 v2.0.14
build github.com/bluenviron/mediamtx: cannot load crypto/ecdh: malformed module path "crypto/ecdh": missing dot in first path element

Building targets
build .: cannot find module for path .

Cleaning module download cache (golang/go#27455)

make[2]: *** [Makefile:39: /etc/openwrt-sdk-19.07.8-ath79-nand_gcc-7.5.0_musl.Linux-x86_64/build_dir/target-mips_24kc_musl/mediamtx-v0.23.2/.built] Error 1
make[2]: Leaving directory '/etc/openwrt-sdk-19.07.8-ath79-nand_gcc-7.5.0_musl.Linux-x86_64/package/mediamtx'
time: package/mediamtx/compile#6.98#6.73#16.77
make[1]: *** [package/Makefile:113: package/mediamtx/compile] Error 2
make[1]: Leaving directory '/etc/openwrt-sdk-19.07.8-ath79-nand_gcc-7.5.0_musl.Linux-x86_64'
make: *** [/etc/openwrt-sdk-19.07.8-ath79-nand_gcc-7.5.0_musl.Linux-x86_64/include/toplevel.mk:220: package/mediamtx/compile] Error 2
```

Does anyone have a tip what to do? Is there something wrong with the crypto/ecdh?

Thanks,
Benj

**Comment by pimdkrisf3** [comment]
Does anyone here have any ideas? I'm stuck and could really use any hints or suggestions. I would greatly appreciate any help. Thank you.

**Comment by aler9** [comment]
Hello, this issue is caused by the fact that OpenWRT 19 ships with a Golang release that is too old and doesn't contain the library "crypto/ecdh". You can either:

* switch to a newer OpenWRT version
* Use your x86 PC to download the server source code and cross-compile a static binary for MIPS:

   ```
   CGO_ENABLED=0 GOOS=linux GOARCH=mipsle  GOMIPS=softfloat go build .
   ```

   the resulting binary can be uploaded to the router and used in place of the package.

[a comment by an automated account is not shown]
